bash-preexec.sh

```diff
@@ -47,7 +47,17 @@ __bp_last_argument_prev_command="$_"
 
 # Command to set our preexec trap. It's invoked once via
 # PROMPT_COMMAND and then removed.
-__bp_trap_install_string="trap '__bp_preexec_invoke_exec \"\$_\"' DEBUG;"
+__bp_trap_install() {
+  local prior_trap=$(trap -p DEBUG | sed "s/[^']*'\(.*\)'[^']*/\1/")
```

Comment:
Modified your change via my comment below then found this issue.

I think there's an edge case here where a trap contains output. If I set a simple debug trap with an echo it's pulling in the output that is produced from the debug trap here. and setting it in the original debug trap function. e.g:

```bash
$ trap
trap -- '' SIGTSTP
trap -- '' SIGTTIN
trap -- '' SIGTTOU

$ trap 'echo "from trap"' DEBUG

$ source bash-preexec.sh 
from trap
from trap
from trap
from trap
from trap
from trap
from trap
from trap
from trap

$ type __bp_original_debug_trap
from trap
__bp_original_debug_trap is a function
__bp_original_debug_trap () 
{ 
    from trap;
    from trap;
    echo "from trap"
}
```

This causes __bp_original_debug_trap to error with the output 'from trap' in the function.